document/lib/Pdf/Document/FontDict.hs

```diff
@@ -285,32 +285,43 @@ loadFontDescriptor pdf fontDict = do
         , fdCharSet = charSet
         }
   where
-    required = requiredInDict "FontDescriptor"
-    optional = optionalInDict "FontDescriptor"
+    required = requiredInDict pdf "FontDescriptor"
+    optional = optionalInDict pdf "FontDescriptor"
     nameValue' = fmap Name.toByteString . nameValue
 
 -- | Parse a value from a required field of a dictionary. This will
 -- raise an exception if a) the field is not present or b) the field
 -- value has a false type.
-requiredInDict :: String -- ^ a context for a failure notice
+requiredInDict :: Pdf    -- ^ in case the field is a reference
+               -> String -- ^ a context for a failure notice
                -> Name   -- ^ name of dictionary field
                -> (Object -> Maybe a) -- ^ function for type-casting the object
                -> Dict                -- ^ the dictionary
                -> IO a
-requiredInDict context key typeFun =
+requiredInDict pdf context key typeFun =
   join .
   sure . (`notice` (context ++ ": " ++ msg ++ " should exist")) .
-  fmap (sure . (`notice` (context ++ ": " ++ msg ++ " type failure")) . typeFun) .
+  fmap
+    (join .
+     fmap (sure . (`notice` (context ++ ": " ++ msg ++ " type failure")) . typeFun) .
+     followRef pdf) .
   HashMap.lookup key
   where
     msg = Text.unpack $ decodeUtf8With ignore $ Name.toByteString key
 
+followRef :: Pdf -> Object -> IO Object
+followRef pdf (Ref ref) = lookupObject pdf ref
+followRef _ o = return o
```

Comment:
Use `deref` from [here](https://github.com/Yuras/pdf-toolbox/blob/74fcc6492a466ea24007d7a839367bf133150444/document/lib/Pdf/Document/Pdf.hs#L138-L140) instead

Reply:
Done! Sorry for missing that.